lang/perl/patches/310-recompile-hack.patch

```diff
@@ -5,12 +5,12 @@ patch has been reported to work around that issue.
 
 --- a/Makefile.SH
 +++ b/Makefile.SH
-@@ -818,7 +818,7 @@ $(LIBPERL): $& $(obj) $(DYNALOADER) $(LI
+@@ -867,7 +867,7 @@ $(LIBPERL): $& $(perllib_objs) $(DYNALOA
  	true)
  		$spitshell >>$Makefile <<'!NO!SUBS!'
  	rm -f $@
--	$(LD) -o $@ $(SHRPLDFLAGS) $(obj) $(DYNALOADER) $(libs)
-+	$(CC) -o $@ $(SHRPLDFLAGS) $(obj) $(DYNALOADER) $(libs)
+-	$(LD) -o $@ $(SHRPLDFLAGS) $(perllib_objs) $(DYNALOADER) $(libs)
++	$(CC) -o $@ $(SHRPLDFLAGS) $(perllib_objs) $(DYNALOADER) $(libs)
```

Comment:
My answer is: I don't know.

The patch was made with the following comment:

> Apparently, recompiling/relinking fails under some circumstances. This
> patch has been reported to work around that issue.
> 
> Signed-off-by: Marcel Denia <naoir@gmx.net>

I could not find Marcel Denia on github so I doubt we will get any answers.

The problem is that I do not know what "circumstances" he was talking about, so I cannot test this.

I would imagine that changing LD= would affect the compile on a more global scale than the patch.

We can try removing the patch completely and see if the problem comes back.